pkg/apis/core/v1beta1/types_seed.go

```diff
@@ -270,8 +270,23 @@ type SeedSettings struct {
 
 // SeedSettingExcessCapacityReservation controls the excess capacity reservation for shoot control planes in the seed.
 type SeedSettingExcessCapacityReservation struct {
-	// Enabled controls whether the excess capacity reservation should be enabled.
+	// Enabled controls whether the default excess capacity reservation should be enabled.
+	// Deprecated: This field is deprecated and will be removed in a future version of Gardener. Please use `Configs` instead.
 	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
+	// Configs configures excess capacity reservation deployments for shoot control planes in the seed.
+	Configs []SeedSettingExcessCapacityReservationConfig `json:"configs,omitempty" protobuf:"bytes,2,rep,name=configs"`
+}
+
+// SeedSettingExcessCapacityReservationConfig configures excess capacity reservation deployments for shoot control planes in the seed.
+type SeedSettingExcessCapacityReservationConfig struct {
+	// Resources specify the resource requests and limits of the excess-capacity-reservation pod.
+	Resources corev1.ResourceList `json:"resources,omitempty" protobuf:"bytes,1,rep,name=resources"`
```

Comment:
I see that `corev1.ResourceList` usages in K8s have the following protobuf tags `casttype=ResourceList,castkey=ResourceName`.
https://github.com/gardener/gardener/blob/7ee780e399287da45cde03150775014affb10165/vendor/k8s.io/api/core/v1/types.go#L2307-L2316

```suggestion
	Resources corev1.ResourceList `json:"resources,omitempty" protobuf:"bytes,1,rep,name=resources,casttype=ResourceList,castkey=ResourceName"`
```